dnf5/commands/download/download.cpp

```diff
@@ -179,6 +194,24 @@ void DownloadCommand::run() {
         return;
     }
 
+    if (source_option->get_value()) {
+        std::map<std::string, libdnf5::rpm::Package> source_pkgs;
+
+        for (auto & [nevra, pkg] : download_pkgs) {
+            libdnf5::rpm::PackageQuery pkg_query(full_pkg_query);
+            pkg_query.resolve_pkg_spec(pkg.get_source_name(), {}, true);
```

Comment:
Resolving just the source name is not enough because the user can specify an exact version to download but this will always download the latest srpm.

You can check out what does old dnf do: https://github.com/rpm-software-management/dnf-plugins-core/blob/master/plugins/download.py#L268-L280